### PR TITLE
ref(tests): Clean up tests with get_valid_response

### PR DESCRIPTION
**Pull request by mgaeta**
We have a really nice helper `get_valid_response` that combines parsing parameters, making a request, and validating the response. It makes API tests much simpler.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_projects.py [hunk 1]
```diff
@@ -1,167 +1,170 @@
...
 
-        response = self.client.get(f"{self.path}?statsPeriod=", format="json")
+        response = self.get_valid_response(self.organization.slug, qs_params={"statsPeriod": ""})
         self.check_valid_response(response, projects)
         assert "stats" not in response.data[0]
 
-        response = self.client.get(f"{self.path}?statsPeriod=48h", format="json")
-        assert response.status_code == 400
+        self.get_valid_response(
+            self.organization.slug, qs_params={"statsPeriod": "48h"}, status_code=400
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_projects.py:59]: Could `qs_params` be `data` or `query` to align with the `self.client` API. Having to remember different names can be taxing.

File: tests/sentry/api/endpoints/test_organization_teams.py [hunk 2]
```diff
@@ -125,36 +130,27 @@ def test_valid_params(self):
...
-
-        assert resp.status_code == 201, resp.content
-
-        resp = self.client.post(self.path, data={"name": "hello world", "slug": "foobar"})
-
-        assert resp.status_code == 409, resp.content
+        self.get_valid_response(
+            self.organization.slug, name="hello world", slug="foobar", status_code=201
+        )
+        self.get_valid_response(
```
Comment (markstory) [inline tests/sentry/api/endpoints/test_organization_teams.py:149]: Should the method be `get_response`? The name `get_valid_response()` and expecting a 409 seem at odds.

File: tests/sentry/api/endpoints/test_organization_projects.py (hunk 1 above)
Comment (mgaeta) [reply tests/sentry/api/endpoints/test_organization_projects.py:59]: The problem is `**params` are directly transformed to the POST data so we can't use any good names without collisions with the keys we want to test. 
Also, I agree that different names is a mentally taxing hurdle, I keep getting `status_code` and `response_code` mixed up when I write tests.

File: tests/sentry/api/endpoints/test_organization_teams.py (hunk 2 above)
Comment (mgaeta) [reply tests/sentry/api/endpoints/test_organization_teams.py:149]: `get_response` is already used, but I've created two new functions (`get_success_response` and `get_error_response`) that should make the tests clearer.

**Review by markstory — APPROVED** [review]
Great stuff. Love to see less boilerplate in tests.

**Pull request merged**